src/ai/backend/manager/models/scaling_group.py

```diff
@@ -315,6 +320,42 @@ class ScalingGroup(graphene.ObjectType):
     scheduler_opts = graphene.JSONString()
     use_host_network = graphene.Boolean()
 
+    # Dynamic fields.
+    agent_count = graphene.Int(description="Added in 24.03.5.")
+    agent_total_resource_slots = graphene.JSONString(description="Added in 24.03.5.")
+
+    async def resolve_agent_count(self, info: graphene.ResolveInfo) -> int:
+        from .agent import Agent
+
+        return await Agent.load_count(
+            info.context,
+            scaling_group=self.name,
+        )
+
+    async def resolve_agent_total_resource_slots(
+        self, info: graphene.ResolveInfo
+    ) -> Mapping[str, Any]:
+        from .agent import AgentRow
+
+        graph_ctx = info.context
+        async with graph_ctx.db.begin_readonly_session() as db_session:
+            query_stmt = sa.select([AgentRow.occupied_slots, AgentRow.available_slots]).where(
+                AgentRow.scaling_group == self.name
+            )
+            result = (await db_session.execute(query_stmt)).fetchall()
```

Comment:
```suggestion
            query_stmt = sa.select(AgentRow).where(
                AgentRow.scaling_group == self.name
            ).options(load_only(AgentRow.occupied_slots, AgentRow.available_slots))
            result = (await db_session.scalars(query_stmt)).all()
            result = cast(list[AgentRow], result)
```
We can use more ORM patterns like this

Reply:
I did not understand what you meant "more ORM pattern".
Could you please let me know what advantage we can take from this query?
I'm curious to know when and for what purpose I should use the `options`, and `load_only` instead of just using a select statement.

Reply:
it is not what we must do but it is recommended to use ORM pattern in this system because of the following reasons.
1. It is too hard to distinguish which API to use among `execute()`, `scalar()` and `scalars()`. The result of those APIs are different between SQLAlchemy Core API and ORM API, which can cause error. We can reduce such confusion by using only ORM APIs.
2. Easy to guess type. Using ORM APIs ensure that the result of queries are always ORM objects.

We can simply query like this `sa.select([AgentRow.available_slots, AgentRow. occupied_slots])`.
As I said, it is not what we must do. But strong type checking is what we should do. So, let's apply type casting and let the type-checker checks thoroughly